wallet/walletrpc.c

```diff
@@ -801,11 +801,10 @@ static struct command_result *json_listtransactions(struct command *cmd,
 		json_add_u64(response, "blockheight", txs[i].blockheight);
 		json_add_num(response, "txindex", txs[i].txindex);
 		json_add_txtypes(response, "type", txs[i].type);
-		if (txs[i].channel_id != 0) {
-			json_add_num(response, "channel_id", txs[i].channel_id);
-		} else {
-			json_add_null(response, "channel_id");
-		}
+		if (txs[i].scid)
```

Comment:
imho we should keep the `channel_id` in there, maybe call it a `group` key. Alternatively we could also stop deleting channels from the `channels` table ensuring that the `short_channel_id` will always resolve correctly. If a wallet makes use of the grouping it would suddenly not group them anymore once we delete the channel from the `channels` table.

Reply:
I've added a new PR that keeps the `channels` entries around, so that these references will keep working after the channel is fully settled.